maths/README.md

```diff
@@ -0,0 +1,31 @@
+# Mutiny Math Operators
+
+This library provides a set of _operators_ for [Mutiny](https://smallrye.io/smallrye-mutiny).
+
+You can use these operators using the `plug` method offered by Mutiny:
+
+```java
+Multi<Long> counts = Multi.createFrom().items("a", "b", "c", "d", "e")
+                            .plug(Maths.count())
+```
+
+These operators provide streams (`Multi`) emitting new values when the computed value changes. To get the last value, use `collect().last()`:
+
+```java
+String max = Multi.createFrom().items("e", "b", "c", "f", "g", "e")
+                .plug(Maths.max())
+                .collect().last()
+                .await().indefinitely();
+```
+
+The `io.smallrye.maths.Maths` class provides the entry point to the offered operators:
+
+* count - emits the number of items emitted by the upstream
+* index - emits `Tuple2<Long, T>` for each item from the upstream. The first element of the tuple is the index (0-based), and the second if the item 
+* min / max - emits the min/max item from the upstream, using Java comparator
+* top(x) - emits the top x items from the upstream, a new ranking is emitted every time it changes.
+* sum - emits the sum of all the items emitted by the upstream
+* average - emits the average of all the items emitted by the upstream
```

Comment:
Did you think about adding a `stats` (or `statistics`)  operator that would return the common stats in one go? Similarly to `sum` and `average`, it would be constrained to `T extends Number` and output `double`s. Something like this:

```java
public final class Statistics {
    public final long count;
    public final double min;
    public final double max;
    public final double sum;
}
```

Average is then easy to compute.

Alternatively, there's a relatively simple algorithm for computing the first four moments in a "streaming" fashion: https://www.johndcook.com/blog/skewness_kurtosis/ So those could be added here as well.

Just an idea :-)

Reply:
Definitely.

This is far from being all, there are a few things I want to add, and statistics is one of them (support for Comparator too).

Reply:
BTW, do you believe that adding median would be interesting? I rarely use them personally, but it may be used.

Reply:
I don't do mathS really :-), but I think mean and median are both very useful.

Reply:
Median is always interesting